integration/update_compat_test.go

```diff
@@ -181,87 +181,78 @@ func testUpdateManyCompat(t *testing.T, testCases map[string]testUpdateManyCompa
 
 			t.Parallel()
 
-			providers := shareddata.AllProviders()
-			if tc.providers != nil {
-				providers = tc.providers
+			setupOpts := &setup.SetupCompatOpts{
+				Providers: tc.providers,
 			}
 
-			s := setup.SetupCompatWithOpts(t, &setup.SetupCompatOpts{
-				Providers:                providers,
-				AddNonExistentCollection: true,
-			})
+			if tc.providers == nil {
+				setupOpts.Providers = shareddata.AllProviders()
+				setupOpts.AddNonExistentCollection = true
+			}
+
+			s := setup.SetupCompatWithOpts(t, setupOpts)
+
 			ctx, targetCollections, compatCollections := s.Ctx, s.TargetCollections, s.CompatCollections
 
 			update := tc.update
 			require.NotNil(t, update, "`update` must be set")
 
+			filter := tc.filter
+			if filter == nil {
+				filter = bson.D{}
+			}
+
 			var nonEmptyResults bool
 			for i := range targetCollections {
 				targetCollection := targetCollections[i]
 				compatCollection := compatCollections[i]
 				t.Run(targetCollection.Name(), func(t *testing.T) {
 					t.Helper()
 
-					allDocs := FindAll(t, ctx, targetCollection)
-
-					for _, doc := range allDocs {
-						id, _ := ConvertDocument(t, doc).Get("_id")
-						require.NotNil(t, id)
-
-						t.Run(fmt.Sprint(id), func(t *testing.T) {
-							t.Helper()
-
-							filter := tc.filter
-							if tc.filter == nil {
-								filter = bson.D{{"_id", id}}
-							}
-
-							var targetUpdateRes, compatUpdateRes *mongo.UpdateResult
-							var targetErr, compatErr error
+					var targetUpdateRes, compatUpdateRes *mongo.UpdateResult
+					var targetErr, compatErr error
 
-							targetUpdateRes, targetErr = targetCollection.UpdateMany(ctx, filter, update, tc.updateOpts)
-							compatUpdateRes, compatErr = compatCollection.UpdateMany(ctx, filter, update, tc.updateOpts)
+					targetUpdateRes, targetErr = targetCollection.UpdateMany(ctx, filter, update, tc.updateOpts)
+					compatUpdateRes, compatErr = compatCollection.UpdateMany(ctx, filter, update, tc.updateOpts)
 
-							if targetErr != nil {
-								t.Logf("Target error: %v", targetErr)
-								t.Logf("Compat error: %v", compatErr)
+					if targetErr != nil {
+						t.Logf("Target error: %v", targetErr)
+						t.Logf("Compat error: %v", compatErr)
```

Comment:
Why are you [just] printing targetErr and compatErr if target error is not nil? Is this just a leftover from manual testing?

Reply:
I have modified this function as per [this](https://ferretdb.slack.com/archives/C02NWN1K3MJ/p1702803985556339) discussion on slack. I have kept the existing code as it is. I see that the same pattern has been followed on many of the `compat*` functions in integration test.

Reply:
It makes sense for me. If target returns errors, then we compare it compatErr and return from the test (there is no point in continuing after that). If the assertion fails, then we will have both compatErr and targetErr available in the test output, so that we can debug.